### ISSUE TITLE
Feature request: Allow to create backups of the configuration

### ISSUE THREAD

**Issue by kang-makes** [tuskyapp/Tusky#4242]
I changed my phone and I had to manually configure the few account I had in the fedi plus the configuration about how I would like to see the tabs, where, how would I like the app to behave when it opens, yada, yada...

It would be awesome to have a way to export all the config so I can have a backup or sync it between devices. 

* * * *

- [x] I searched or browsed the repo’s other issues to ensure this is not a duplicate.

**Comment by haverholm** [comment]
I agree, and will add per #4984 that export/import of keyword filters would be very handy!

**Comment by connyduck** [comment]
I agree we want a feature like this, but there are a lot of things to consider. I can see multiple usecases, from easy to hard implementation:

1) Exporting the Tusky local settings of one account. Used e.g. when switching devices, or adding a new account to set ones favourite defaults. Can be a json file or something. Does not contain any account information.
2) Exporting everything Tusky including all accounts and their access token. Used e.g. when switching devices. One import and everything is restored. Must support encryption as there is sensitive data involved.
3) Export everything from one account including server side settings, and maybe existing list, filters, etc. Used e.g. when setting up a new Mastodon account to copy everything over as an addition to account migration. Can be very tricky when the importing server does not support a certain feature or is populated with conflicting data.

Which general direction should this go?

**Comment by haverholm** [comment]
My two cents, as a user who has both changed devices and instances within the past year — start with the simpler task. Bear in mind, I'm not a developer.

A) Model 1 fits my current use case post-instance migration 🙂 I have a comprehensive set of keyword filters that I don't care to replicate manually to a new account.

B) Maybe a less involved approach to models 2 and 3 will turn out to be the way to go. If you leave out the access token from 2, perhaps you won't need the encryption? Requiring a new session from a new device is probably safer anyway.

**Comment by jahway603** [comment]
I was about to write up a new feature request for this too, but then found this one. I like implementation 1, but expand it so you can do this for each configured in Tusky account separately.
